train.py

```diff
@@ -1 +1,218 @@
+
+    :param checkpointer:
+        Object containing parameters needed in model saving.
+
+    :param tensorboard:
+        Object containing parameters needed in logs saving.
+
+    :param x_train:
+        Input training data.
+
+    :param y_train:
+        Output training data.
+
+    :param x_validate:
+        Input validation data.
+
+    :param y_validate:
+        Output validation data.
+
+    :return:
+        Trained neural network model.
+    """
+    model.fit(x_train, y_train,
+              batch_size=batch_size, epochs=epochs, verbose=1, shuffle=False,
+              validation_data=(x_validate, y_validate), callbacks=[checkpointer, tensorboard])
+
+    return model
+
+
+def testing(epochs, saving_period, x_test, y_test, args, model):
+    """
+    This module tests model after each training epoch on testing dataset and shows score.
+
+    :param epochs:
+        Number of training epochs.
+
+    :param saving_period:
+        How many epochs of training must pass before saving model.
+
+    :param x_test:
+        Input testing data.
+
+    :param y_test:
+        Output testing data.
+
+    :param args:
+        Path for loading model.
+
+    :param model
+        Tested  neural network model.
+    """
+    for i in range(int(epochs/saving_period)):
+        # Downloading weights
+        try:
+            model = load_model(args.model_load.format(i+1))
+        except:
+            print("Unable to load model")
+        # Checking results with testing dataset
+        score_test = model.evaluate(x_test, y_test, verbose=1)
+        print("Score after {}. epoch on testing dataset: ".format(i+1))
+        print("Loss:")
+        print(score_test[0])
+        print("Accuracy:")
+        print(score_test[1])
+
+
+def parse_args():
+    """
+    This module parses command line arguments.
+
+    :return:
+        Parser containing command line data.
+    """
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument('--dataset', '-d', metavar="csv file containing dataset",
+                        help='input file containing dataset required in training', default='smile_warrior_dataset.csv')
+
+    parser.add_argument('--batch_size', '-b', metavar="number of images in one batch",
+                        help='number of images in one batch', type=int, default=32)
+
+    parser.add_argument('--epochs', '-e', metavar="number of training epochs",
+                        help='number of training epochs', type=int, default=10)
+
+    parser.add_argument('--saving_period', '-s', metavar="how many epochs must pass before saving",
+                        help='how many epochs must pass before saving', type=int, default=1)
+
+    parser.add_argument('--learning_rate', '-lr', metavar="specifies learning rate",
+                        help='specifies learning rate', type=float, default=0.0001)
+
+    parser.add_argument('--models_dir', '-m', metavar="specifies saving directory for models",
```

Comment:
Rather use - than _

Reply:
Are you sure? I get errors when I try to do this that way.